server/src/modules/users/domain/entities/auth-code/auth-code.ts

```diff
@@ -0,0 +1,38 @@
+import { Result } from '../../../../../shared/core/result'
+import { AggregateRoot } from '../../../../../shared/domain/aggregate-root'
+import { UniqueEntityID } from '../../../../../shared/domain/unique-entity-id'
+import { AuthCodeCreated } from '../../events/auth-code-created'
+import { AuthCodeString } from '../../value-objects/auth-code'
+
+interface AuthCodeProps {
+  clientId: string,
+  authCode: AuthCodeString
+}
+
+export class AuthCode extends AggregateRoot<AuthCodeProps> {
+  public static create(props: AuthCodeProps, id?: UniqueEntityID): Result<AuthCode, Error> {
+    const isNewAuthCode = !!id === false
```

Comment:
does this just do an explicit check to see if the `id` field is falsy? (`null`, `undefined`, `""`)

Reply:
Yep, I believe so. It's basically sanitizing other falsy values to `false`.

Reply:
I'm pretty sure the only falsy value you could get here is `undefined`.
```suggestion
    const isNewAuthCode = (id === undefined)
```

Reply:
this works